src/java_bytecode/generate_java_generic_type.cpp

```diff
@@ -323,21 +267,36 @@ irep_idt generate_java_generic_typet::build_generic_name(
       "All implicit generic types must be concretised");
     auto implicit_generic_type_p =
       implicitly_generic_original_class.implicit_generic_types().begin();
-    const auto &implicit_generic_types_end =
-      implicitly_generic_original_class.implicit_generic_types().end();
     std::string current_outer_class_name;
 
-    while(implicit_generic_type_p != implicit_generic_types_end)
+    while(implicit_generic_type_p !=
```

Comment:
These nested loops are quite complicated; suggest adding comments including a short example of what's going on.